ptlshs/conn.go

```diff
@@ -94,6 +94,10 @@ func Client(toServer net.Conn, cfg DialerConfig) Conn {
 	return &clientConn{toServer, cfg, nil, nil, newOnce(), newOnce()}
 }
 
+func (c *clientConn) Wrapped() net.Conn {
+	return c.Conn
```

Comment:
If we're going to allow outside callers to read this value, we need to add a lock.  This is because `c.Conn` is written to during `c.handshake`.

Reply:
Good catch, will fix, thanks!

Reply:
Actually, I'm a bit confused as to why this would be a new problem? The existing code accesses this in a variety of places...

Reply:
The handshake is locked down via `c.shakeOnce`.  This synchronization protection is called in `c.Read` and `c.Write` as well.  This ensures that `c.handshake` is only run once and that no I/O is happening in parallel.  Once the handshake is complete, `c.Conn` is finalized and can be treated as read-only, safe to access concurrently.  Since `c.Wrapped` could be called during the handshake, it may access `c.Conn` while `c.Conn` is written to, introducing a new race condition.

All that said, `c.Close` has the same problem as do the `Set*Deadline` functions from the embedded `net.Conn`.  The most correct thing to do here would be to introduce a `sync.Mutex` covering `c.Conn`, use that in `c.Close` and `c.Wrapped`, and add definitions for `c.Set*Deadline` which also use the mutex.

All of this mess is because I chose to overwrite the embedded `Conn`.  Looking at the code with fresh eyes, I think I can just avoid that in the first place and simplify this.  Let me work on that real quick and send you a PR.

Reply:
Sorry, taking a bit longer than expected.  I'll get a PR to you tomorrow.

Reply:
I've actually got a solution, I'll push it shortly.

Reply:
Solution pushed, just uses a lock on clientConn like we do on serverConn.